src/headless/plugins/chat/model.js

```diff
@@ -78,9 +79,24 @@ const ChatBox = ModelWithContact.extend({
          * @example _converse.api.listen.on('chatBoxInitialized', model => { ... });
          */
         await api.trigger('chatBoxInitialized', this, {'Synchronous': true});
+
+        const blocking_supported = await api.blocking?.supported();
+        if (blocking_supported) {
+            await api.waitUntil("blockListFetched");
+            this.checkIfContactBlocked(api.blocking.blocklist());
```

Comment:
`await api.waitUntil("blockListFetched")` should be called inside `api.blocking.blocklist()`.

We don't want consumers of our API to need to know that they have to first wait for the `blockListFetched` promise. That's an implementation detail that we need to handle inside the API.